CMakeLists.txt

```diff
@@ -1,179 +1,141 @@
-        LINKER:--wrap=__realloc_r
-        LINKER:--wrap=__memalign_r
-        LINKER:--wrap=__calloc_r
-        LINKER:--wrap=exit
-        LINKER:--wrap=atexit
-        LINKER:-n
-        )
-set(LINKER_SYS_LIBS
-        LINKER:--start-group
-        -lstdc++
-        -lsupc++
-        -lm
-        -lc
-        -lgcc
-        -lnosys
-        LINKER:--end-group
-        )
-set(LINKER_SCRIPT_SOURCE_PATH "${CMAKE_SOURCE_DIR}/mbed-os/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F446xE/TOOLCHAIN_GCC_ARM/STM32F446XE.ld")
-set(LINKER_SCRIPT_OUTPUT_PATH "${CMAKE_CURRENT_BINARY_DIR}/STM32F446XE.out.ld")
-add_custom_command(OUTPUT ${LINKER_SCRIPT_OUTPUT_PATH}
-        COMMAND ${CMAKE_CPP} -E -P ${PROCESSOR_FLAGS} ${MEMORY_DEFINITIONS_FOR_LINKER} ${LINKER_SCRIPT_SOURCE_PATH} -o ${LINKER_SCRIPT_OUTPUT_PATH}
-        MAIN_DEPENDENCY ${LINKER_SCRIPT_SOURCE_PATH}
-        COMMENT "Preprocessing linker script"
-        )
-add_custom_target(linker-script DEPENDS ${LINKER_SCRIPT_OUTPUT_PATH})
-add_link_options(-T${LINKER_SCRIPT_OUTPUT_PATH} ${LINKER_FLAGS} ${PROCESSOR_FLAGS} ${MEMORY_DEFINITIONS_FOR_LINKER} ${LINKER_SYS_LIBS})
-
-# use objcopy to create .bin files out of ELF files
-function(target_bin_from_elf TARGET-NAME)
-    string(REPLACE ".elf" "" BASE-TARGET-NAME ${TARGET-NAME})
-    add_custom_command(TARGET ${TARGET-NAME}
-            POST_BUILD
-            COMMAND ${CMAKE_OBJCOPY} -O binary ${TARGET-NAME} ${BASE-TARGET-NAME}.bin
-            COMMENT "Generating bin from elf")
-    set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES ${BASE-TARGET-NAME}.bin)
-endfunction()
+# Force extra compile options (hack until https://github.com/ARMmbed/mbed-tools/issues/261 is fixed)
+add_compile_options(
+        -Wpedantic
+        -Wvla
+        -Werror)
 
-# link against mbed and precompile headers
-function(target_set_mbed_dependency TARGET-NAME)
-    target_link_libraries(${TARGET-NAME} PRIVATE mbed-os)
-    target_precompile_headers(${TARGET-NAME} REUSE_FROM mbed-os)
-endfunction()
+# Generate compile commands database for other tools to use
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE INTERNAL "")
 
-# link against mbed and setup linker script
-function(target_set_firmware_properties TARGET-NAME)
-    target_set_mbed_dependency(${TARGET-NAME})
+# Include mbed os cmake logic
+include(${MBED_PATH}/tools/cmake/app.cmake)
 
-    set_target_properties(${TARGET-NAME} PROPERTIES LINK_DEPENDS ${LINKER_SCRIPT_OUTPUT_PATH})
-    add_dependencies(${TARGET-NAME} linker-script)
+# Force colored output from compiler
+add_compile_options(-fdiagnostics-color)
 
-    target_bin_from_elf(${TARGET-NAME})
-endfunction()
+# Include custom targets
+add_subdirectory(${TARGETS_DIR}/TARGET_ARM_REV2)
+add_subdirectory(${TARGETS_DIR}/TARGET_GAMEPAD_REV1)
+add_subdirectory(${TARGETS_DIR}/TARGET_GIMBAL_REV2)
+add_subdirectory(${TARGETS_DIR}/TARGET_PDB_REV1)
+add_subdirectory(${TARGETS_DIR}/TARGET_SCIENCE_REV2)
+add_subdirectory(${TARGETS_DIR}/TARGET_UWRT_NUCLEO)
```

Comment:
Yay rev specific targets!!

Reply:
Also can you rename to PDB_REV2? PDB rev1 was the one without a mcu on it

Reply:
done